src/custom/state/fee/middleware.ts

```diff
@@ -0,0 +1,44 @@
+import { isAnyOf, Middleware } from '@reduxjs/toolkit'
+import { getFee } from 'utils/fees'
+import { replaceSwapState, selectCurrency, switchCurrencies } from '@src/state/swap/actions'
+import { updateFee } from './actions'
+import { FeeInformationState } from './reducer'
+import { SwapState } from '@src/state/swap/reducer'
+
+const isCurrencyChangeAction = isAnyOf(selectCurrency, replaceSwapState, switchCurrencies)
+const isSelectCurrency = isAnyOf(selectCurrency)
+const isReplaceState = isAnyOf(replaceSwapState)
+
+export const applyFeeMiddleware: Middleware = ({ dispatch, getState }) => next => async action => {
+  if (isCurrencyChangeAction(action)) {
+    const {
+      swap: {
+        OUTPUT: { currencyId }
+      },
+      fee: { feesMap }
+    }: { fee: FeeInformationState; swap: SwapState } = getState()
```

Comment:
Maybe would look better with less destructuring,or destructure in steps

Reply:
ok, done